src/services/CalculatorProcess.js

```diff
@@ -1,51 +1,52 @@
-let memory = [];
-let result = "";
-function calculatorProcess(content,visor) {
-	
-	const numbersOperators = () => String(eval((result)));
-	
-	memory += content;
+let memory = "";
+let sLength;
 
-	if (memory.includes("=")) {
+function CalculatorProcess(content, visor) {
 
-		result = memory.slice(0,-1);
+	const firstInteraction = ["%", "/", "*", "+", "-", "S", "C", "="];
 
-		if (memory.includes("+")) {
-			memory = numbersOperators();
-		}
+	const conversion = (string) => String(eval(string));
 
-		if (memory.includes("-")) {
-			memory = numbersOperators();
-		}
+	memory += content;
 
-		if (memory.includes("*")) {
-			memory = numbersOperators();
-		}
+	if ((firstInteraction.includes(memory[0]) && memory.length < 2) || memory.includes("C")) {
+		memory = "";
 
-		if (memory.includes("/")) {
-			memory = numbersOperators();
-		}
+	}
+
+	if (memory[0] == ".") {
+		memory = "0.";
+	}
+
+	if (memory.includes("=")) {
+
+		memory = memory.slice(0, -1);
 
 		if (memory.includes("%")) {
-			memory =
-    String(eval((memory[0]+memory[1]))*("0.0"+memory[3]));
+
+			memory = memory.replace(/[%]/g, "/100*");
 		}
-	}
 
-	if (memory.includes("C")) {
-		memory = "";
+		memory = conversion(memory)
 	}
 
-	if (memory.includes("+/-")) {
-		memory = String(eval((memory[0])*-1));
+	if (memory.includes("S")) { //S means Signal for change number's signal
```

Comment:
you can create a constant for this string, to decribe what this `S` means

for example: `MAINUS_PLUS_SIGNAL = 'S'`

Reply:
hmm yea could be

Reply:
done